### PR TITLE
add extremely rudimentary emacs mode

### LINKED ISSUE #393: Basic Emacs Mode
We should have a really basic emacs mode with highlighting and an output pane. I can just copy and paste some stuff I've done in the past for JonPRL etc.

### PR DESCRIPTION
**Pull request by jonsterling**
this is a really crappy emacs mode! Someone who is an emacs hacker is welcome to replace it.

Resolves #393

### CODE REVIEW

File: emacs/redtt.el [hunk 1]
```diff
@@ -0,0 +1,187 @@
+;;; redtt.el --- Major mode for editing redtt proofs and interacting with redtt  -*- lexical-binding: t; -*-
+
+;; Copyright (C) 2016  Jonathan Sterling
+;; Copyright (C) 2017  Jonathan Sterling, Favonia
```
Comment (favonia) [inline emacs/redtt.el]: Maybe we should update this to 2018. lol

Comment (jonsterling) [reply emacs/redtt.el]: lol

File: emacs/redtt.el [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+(defvar redtt-mode-syntax-table
+  (let ((table (make-syntax-table)))
+    (modify-syntax-entry ?- "w" table)
+    (modify-syntax-entry ?_ "w" table)
+    (modify-syntax-entry ?= "w" table)
+    (modify-syntax-entry ?' "w" table)
+    (modify-syntax-entry ?/  "_ 123" table)
+    (modify-syntax-entry ?\n ">" table)
+    table)
```
Comment (jonsterling) [inline emacs/redtt.el]: LOL who knows what any of this crap means. God, I hate emacs so much.

File: emacs/redtt.el [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+                 ;; Emacs compile config stuff - these are special vars
+                 (compilation-buffer-name-function
+                  'redtt--compilation-buffer-name-function)
+                 (default-directory dir))
+            (compile command)))
+      (error "Buffer has no file name"))))
+
+
+(defconst redtt--revolutionary-sayings
+  '("Maybe The Formalist Clique Isn't So Bad After All!"
```
Comment (favonia) [inline emacs/redtt.el]: while this might be true, it's not Revolutionary. I vote for removing this.

**Comment by jonsterling** [comment]
After this is merged in, I'll submit to melpa.

**Pull request merged**